x-pack/plugins/security_solution/cypress/objects/rule.ts

```diff
@@ -498,14 +500,18 @@ export const expectedExportedRule = (ruleResponse: Cypress.Response<RulesSchema>
     name,
     risk_score: riskScore,
     severity,
-    query,
     tags,
     timeline_id: timelineId,
     timeline_title: timelineTitle,
   } = ruleResponse.body;
 
+  let query: string | undefined;
+  if (ruleResponse.body.type === 'query') {
+    query = ruleResponse.body.query;
+  }
+
   // NOTE: Order of the properties in this object matters for the tests to work.
-  const rule: RulesSchema = {
```

Comment:
There's no type in the schemas at the moment that represents this object accurately because the object does not specify values (even `undefined`) for all fields. The response schema requires all fields to be explicitly specified to ensure that the conversion between internal and response schemas doesn't miss any fields (https://github.com/elastic/kibana/pull/137628 added this requirement) but the side effect is that mock objects now have to specify `undefined` for many of the fields. As a follow up, the next step is probably to build 2 variants of the response schema, one for use in conversions and one for convenience when building mock objects so `undefined` fields can be left out.

I'll add a TODO comment here with a link to this PR so we can identify all the places where types are missing in the follow up and add them back.

Reply:
Actually I think we can use `Partial<FullResponseSchema>` here, which isn't great but is better than nothing until we can get a proper and convenient type for this sort of object construction.

Reply:
@marshallmain 🙏